pkg/awsclient/client.go

```diff
@@ -102,16 +102,18 @@ type Client interface {
 	RevokeSecurityGroupIngress(*ec2.RevokeSecurityGroupIngressInput) (*ec2.RevokeSecurityGroupIngressOutput, error)
 	// DescribeSubnets to find subnet for master nodes for incoming elb
 	DescribeSubnets(*ec2.DescribeSubnetsInput) (*ec2.DescribeSubnetsOutput, error)
+	// CreateTags to apply tags to EC2 resources
+	CreateTags(*ec2.CreateTagsInput) (*ec2.CreateTagsOutput, error)
 }
 
-type awsClient struct {
+type AwsClient struct {
```

Comment:
Do you want this to be exported? Is there a use case where this client will need to be built outside of a call to `NewClient`?

Reply:
It used to be unexported, and I ran into issues in development with that, so I changed it to be exported. Sadly, I can't recall the specifics. Clearly prior to e6dc2a05cac95dc3b72a7928bc3b33bb79881dfa it wasn't exported. For the time being I'll leave it exported.

Reply:
I remembered. It was the introduction of https://github.com/lisa/cloud-ingress-operator/blob/e1cfa77397d48876c0009f301b4509af9a781baf/pkg/awsclient/elb_helper.go#L12-L17 that required exporting. This ties into your other comment below about unified structs.